prow/pjutil/pjutil.go

```diff
@@ -27,8 +29,30 @@ import (
 	"k8s.io/test-infra/prow/kube"
 )
 
+const (
+	jobNameLabel = "prow.k8s.io/spec.name"
+	jobTypeLabel = "prow.k8s.io/spec.type"
+	orgLabel     = "prow.k8s.io/spec.refs.org"
+	repoLabel    = "prow.k8s.io/spec.refs.repo"
+	pullLabel    = "prow.k8s.io/spec.refs.pull"
+)
+
 // NewProwJob initializes a ProwJob out of a ProwJobSpec.
 func NewProwJob(spec kube.ProwJobSpec, labels map[string]string) kube.ProwJob {
+	allLabels := map[string]string{
+		jobNameLabel: spec.Job,
+		jobTypeLabel: string(spec.Type),
+	}
+	if spec.Type != kube.PeriodicJob {
+		allLabels[orgLabel] = spec.Refs.Org
+		allLabels[repoLabel] = spec.Refs.Repo
+		if len(spec.Refs.Pulls) > 0 {
+			allLabels[pullLabel] = strconv.Itoa(spec.Refs.Pulls[0].Number)
+		}
+	}
+	for key, value := range labels {
+		allLabels[key] = value
```

Comment:
It looks like you don't do anything with allLabels? Should you replace L61 with `Labels: allLabels`?

And if this is the case I suspect we don't have the right test coverage?

Reply:
Yes and yes :)

Reply:
Yes, things have been a little bit off this week as you know.

Reply:
These comments are mostly a means to remove the PR from the TODO list on my [pr dashboard](https://k8s-gubernator.appspot.com/pr/fejta), which doesn't work when you reply 😀